kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsSystem.scala

```diff
@@ -58,8 +60,8 @@ class MetricsSystem extends CompositeService("MetricsSystem") {
     meter.mark(value)
   }
 
-  def getGauge[T](name: String): Option[Gauge[T]] = {
-    Option(registry.gauge(name))
+  def getGauge(name: String): Option[Gauge[_]] = {
+    registry.getGauges((metricsName, _) => { metricsName == name }).asScala.map(_._2).headOption
```

Comment:
semantic is correct, but performance is poor, is it possible to have an O(1) approach? if impossible and perf is acceptable, please leave a brief comment to explain that.